### PR TITLE
Consider all labels by default when deriving security identity

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

File: daemon/docker_watcher.go [hunk 1]
```diff
@@ -183,20 +183,13 @@ func (d *Daemon) fetchK8sLabels(dockerLbls map[string]string) (map[string]string
...
 func (d *Daemon) getFilteredLabels(allLabels map[string]string) labels.Labels {
-	var ciliumLabels, k8sLbls, k8sSpecialLabels labels.Labels
+	var ciliumLabels, k8sLbls labels.Labels
 	if podName := k8sDockerLbls.GetPodName(allLabels); podName != "" {
 		k8sNormalLabels, err := d.fetchK8sLabels(allLabels)
 		if err != nil {
 			log.Warningf("Error while getting kubernetes labels: %s", err)
 		} else if k8sNormalLabels != nil {
 			k8sLbls = labels.Map2Labels(k8sNormalLabels, k8s.LabelSource)
-
```
Comment (aanm) [inline daemon/docker_watcher.go:193]: We still need to store the `io.kubernetes.pod.namespace` here.

File: pkg/labels/filter.go [hunk 2]
```diff
@@ -114,18 +114,11 @@ func (cfg *LabelPrefixCfg) Append(l *LabelPrefix) {
...
-			},
-			{
-				Prefix: "io.cilium.",
-			},
-			{
-				Prefix: k8s.PodNamespaceLabel,
-				Source: k8s.LabelSource,
+				Ignore: true,
+				Prefix: "io.kubernetes",
 			},
```
Comment (aanm) [inline pkg/labels/filter.go]: Also ignore `k8s.io/kubernetes/pkg/apis/extensions/DefaultDeploymentUniqueLabelKey` or `"pod-template-hash"`

**Review by aanm — CHANGES_REQUESTED** [review]

File: daemon/docker_watcher.go (hunk 1 above)
Comment (tgraf) [reply daemon/docker_watcher.go:193]: Fixed

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
For future reference:
```
$ cilium endpoint list
ENDPOINT   POLICY        IDENTITY   LABELS (source:key[=value])                   IPv6                     IPv4            STATUS   
           ENFORCEMENT                                                                                                              
173        Enabled       258        k8s:io.kubernetes.pod.namespace=default       fd02::c0a8:210b:0:ad     10.11.42.252    ready    
                                    k8s:k8s-app.guestbook=redis                                                                     
                                    k8s:role=slave                                                                                  
13949      Enabled       259        k8s:io.kubernetes.pod.namespace=default       fd02::c0a8:210b:0:367d   10.11.193.100   ready    
                                    k8s:k8s-app.guestbook=web                                                                       
38939      Enabled       257        k8s:io.kubernetes.pod.namespace=default       fd02::c0a8:210b:0:981b   10.11.148.173   ready    
                                    k8s:k8s-app.guestbook=redis                                                                     
                                    k8s:role=master                                                                                 
56326      Disabled      256        k8s:io.kubernetes.pod.namespace=kube-system   fd02::c0a8:210b:0:dc06   10.11.250.189   ready    
                                    k8s:k8s-app=kube-dns
```

**Pull request merged**